rsbuild.config.ts

```diff
@@ -0,0 +1,125 @@
+import path from "path"
+import nodeExternals from "webpack-node-externals"
+import LoadablePlugin from "@loadable/webpack-plugin"
+import { defineConfig } from "@rsbuild/core"
+import { pluginReact } from "@rsbuild/plugin-react"
+import { pluginAssetsRetry } from "@rsbuild/plugin-assets-retry"
+import { pluginNodePolyfill } from "@rsbuild/plugin-node-polyfill"
+import { EarlyHintsPlugin } from "./rspack/plugins/EarlyHintsPlugin"
+import { loadEnvs } from "@artsy/multienv"
+
+loadEnvs(".env.shared", ".env")
+
+export default defineConfig({
+  environments: {
+    client: {
+      plugins: [pluginAssetsRetry()],
+      source: {
+        entry: {
+          index: "./src/client.tsx",
+        },
+      },
+      output: {
+        target: "web",
+        manifest: true,
+        externals: {
+          async_hooks: "async_hooks", // Required because getAsyncStorage isn't using async import()
+        },
+      },
+      tools: {
+        htmlPlugin: false,
+        rspack: {
+          plugins: [
+            new EarlyHintsPlugin(),
+            new LoadablePlugin({
+              filename: "loadable-stats.json",
+            }),
+          ],
+        },
+      },
+    },
+
+    server: {
+      output: {
+        target: "node",
+        distPath: {
+          root: "dist/server",
+        },
+      },
+      source: {
+        entry: {
+          index: "./src/server",
+        },
+      },
+      tools: {
+        rspack: {
+          externals: [nodeExternals()],
+          node: {
+            __dirname: true,
+          },
+        },
+        swc: {
+          module: {
+            type: "commonjs",
+          },
+        },
+      },
+    },
+  },
+
+  /**
+   * Shared config across environments
+   */
+
+  dev: {
+    progressBar: true,
+    writeToDisk: true,
+  },
+  performance: {
+    // FIXME: Uncomment and play around with different bundle split strategies,
+    // along with stats below:
+    // chunkSplit: {
+    //   strategy: "split-by-size",
+    //   minSize: 30000,
+    //   maxSize: 50000,
+    // },
```

Comment:
Good idea on keeping this commented. I think it might make more sense to test this separately once everything else is stable and in prod to evaluate it properly.